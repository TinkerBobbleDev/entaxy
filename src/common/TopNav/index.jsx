import React from 'react'
import Button from '@material-ui/core/Button'
import { Route, NavLink } from 'react-router-dom'

const TopNav = () => (
  <div>
    <Route
      path="/transactions"
      render={(() => (
        <div>
          <Button size="small" color="inherit" component={NavLink} to="/portfolio">
            Portfolio
          </Button>
        </div>
      ))}
    />
    <Route
      path="/portfolio"
      render={(() => (
        <Button size="small" color="inherit" component={NavLink} to="/transactions">
          Transactions
        </Button>
      ))}
    />
    <Route
      path="/data-sources"
      render={(() => (
        <Button size="small" color="inherit" component={NavLink} to="/transactions">
          Transactions
        </Button>
      ))}
    />
  </div>
)

export default TopNav
