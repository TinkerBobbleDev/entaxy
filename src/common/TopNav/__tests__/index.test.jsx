import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import TopNav from '../'

describe('TopNav', () => {
  it('matches snapshot', () => {
    const component = renderer.create((
      <BrowserRouter>
        <TopNav />
      </BrowserRouter>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
