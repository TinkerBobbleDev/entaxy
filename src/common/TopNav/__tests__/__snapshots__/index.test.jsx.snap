// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav matches snapshot 1`] = `
<div>
  <a
    aria-current={null}
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 MuiButton-sizeSmall-24 MuiButton-colorInherit-22"
    href="/portfolio"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-2"
    >
      Portfolio
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </a>
  <a
    aria-current={null}
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 MuiButton-sizeSmall-24 MuiButton-colorInherit-22"
    href="/import-transactions"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-2"
    >
      Import Transactions
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </a>
</div>
`;
